### ISSUE TITLE
Driver - Slow Submission with 16.7.* (Partial Solution inside)

### ISSUE THREAD

**Issue by Xaymar** [Xaymar/OBS-AMD-Advanced-Media-Framework#18]
Problem: After upgrading to 16.7.3 and downgrading to 16.7.2, submission time is tripled (or more) so that fluid recording becomes impossible.

Solution:
1. Wipe AMD drivers with DDU and install a clean 16.8.2 version. This should return it to normal.
2. Install 1.2.0pre4 or a newer version of the plugin

**Comment by Xaymar** [comment]
This persists even after downgrading, which leads me to believe that 16.7.3 flashes a new VCE bios.

**Comment by InKahootz** [comment]
Shoot, I just upgraded from 16.5.3 or so.

**Comment by Xaymar** [comment]
There are new MFT dlls for h264 and h265 in the 16.7.3 driver which seem to directly interface with the new bios.

**Comment by InKahootz** [comment]
Where are those DLLs located?

**Comment by Xaymar** [comment]
%WINDIR%\System32\DriverStore\FileRepository\c0304500.inf_amd64_9d5a6ee4783926fa\B304449

**Comment by Xaymar** [comment]
After upgrading to 16.7.3, I wrote a bit of code for logging the duration it took to submit frames. I tested it on a blank canvas and got these results:

```
16:39:51.337: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.812194
16:39:51.340: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.508169
16:39:51.347: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.969001
16:39:51.350: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.418002
16:39:51.356: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.198176
16:39:51.359: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.045187
16:39:51.365: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.111991
16:39:51.368: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.788339
16:39:51.375: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.815121
16:39:51.379: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.092404
16:39:51.385: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.061076
16:39:51.387: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.342342
16:39:51.393: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.131902
16:39:51.397: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.910648
16:39:51.403: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.600370
16:39:51.406: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.966398
16:39:51.412: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.853721
16:39:51.415: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.810526
16:39:51.422: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.536657
16:39:51.425: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.673142
16:39:51.431: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.514956
16:39:51.434: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.759043
16:39:51.440: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.603499
16:39:51.444: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.151282
16:39:51.450: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.906627
16:39:51.453: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.516418
16:39:51.459: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.071032
16:39:51.461: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.577571
16:39:51.467: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.985416
16:39:51.471: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.949616
16:39:51.477: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.907765
16:39:51.480: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.138482
16:39:51.486: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.294600
16:39:51.490: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.876231
16:39:51.497: [AMF Encoder 1.2.0pre3] Surface Submission time: 7.028448
16:39:51.500: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.487772
16:39:51.507: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.345798
16:39:51.510: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.871680
16:39:51.516: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.715770
16:39:51.518: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.519547
16:39:51.525: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.219793
16:39:51.528: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.556037
16:39:51.535: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.524710
16:39:51.538: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.692485
16:39:51.544: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.149536
16:39:51.547: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.825885
16:39:51.554: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.340964
16:39:51.556: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.304796
16:39:51.562: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.610243
16:39:51.565: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.081595
16:39:51.571: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.468593
16:39:51.573: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.251322
16:39:51.579: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.932511
16:39:51.582: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.721781
16:39:51.610: [AMF Encoder 1.2.0pre3] Surface Submission time: 28.255728
16:39:51.613: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.812517
16:39:51.620: [AMF Encoder 1.2.0pre3] Surface Submission time: 6.772739
16:39:51.623: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.331534
16:39:51.632: [AMF Encoder 1.2.0pre3] Surface Submission time: 9.210937
16:39:51.635: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.288583
16:39:51.652: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.105212
16:39:51.655: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.062822
16:39:51.671: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.957792
16:39:51.674: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.003090
16:39:51.693: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.352186
16:39:51.697: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.904472
16:39:51.716: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.219354
16:39:51.719: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.431738
16:39:51.728: [AMF Encoder 1.2.0pre3] Surface Submission time: 8.275990
16:39:51.731: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.943073
16:39:51.747: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.775752
16:39:51.750: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.927999
16:39:51.766: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.039710
16:39:51.770: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.524179
16:39:51.788: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.424635
16:39:51.792: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.977857
16:39:51.811: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.326871
16:39:51.814: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.217556
16:39:51.832: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.060555
16:39:51.835: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.972087
16:39:51.854: [AMF Encoder 1.2.0pre3] Surface Submission time: 19.106514
16:39:51.858: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.778182
16:39:51.864: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.951568
16:39:51.867: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.706706
16:39:51.884: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.798588
16:39:51.888: [AMF Encoder 1.2.0pre3] Surface Creation time: 4.119507
16:39:51.907: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.030771
16:39:51.911: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.617475
16:39:51.916: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.770097
16:39:51.919: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.563350
16:39:51.936: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.957305
16:39:51.940: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.189397
16:39:51.955: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.397450
16:39:51.959: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.087852
16:39:51.976: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.498873
16:39:51.980: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.501424
16:39:51.998: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.147391
16:39:52.002: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.895572
16:39:52.018: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.010413
16:39:52.021: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.803415
16:39:52.038: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.044627
16:39:52.041: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.193663
16:39:52.059: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.998548
16:39:52.062: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.253679
16:39:52.078: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.886682
16:39:52.081: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.511866
16:39:52.098: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.977215
16:39:52.102: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.369730
16:39:52.108: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.897241
16:39:52.111: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.364610
16:39:52.117: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.704676
16:39:52.120: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.858596
16:39:52.139: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.235851
16:39:52.142: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.052867
16:39:52.149: [AMF Encoder 1.2.0pre3] Surface Submission time: 7.125726
16:39:52.152: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.348884
16:39:52.167: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.387210
16:39:52.171: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.865707
16:39:52.187: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.384448
16:39:52.190: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.955304
16:39:52.206: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.161936
16:39:52.209: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.377694
16:39:52.224: [AMF Encoder 1.2.0pre3] Surface Submission time: 14.903098
16:39:52.228: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.035232
16:39:52.244: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.074127
16:39:52.248: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.846732
16:39:52.264: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.818702
16:39:52.267: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.134785
16:39:52.283: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.671079
16:39:52.287: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.500855
16:39:52.302: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.083431
16:39:52.306: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.011338
16:39:52.323: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.099524
16:39:52.326: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.026698
16:39:52.343: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.263360
16:39:52.346: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.843237
16:39:52.363: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.072135
16:39:52.366: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.106626
16:39:52.382: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.275508
16:39:52.386: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.362619
16:39:52.392: [AMF Encoder 1.2.0pre3] Surface Submission time: 5.813331
16:39:52.395: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.695329
16:39:52.410: [AMF Encoder 1.2.0pre3] Surface Submission time: 14.654499
16:39:52.414: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.856971
16:39:52.430: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.470917
16:39:52.435: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.496589
16:39:52.452: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.080466
16:39:52.455: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.114306
16:39:52.471: [AMF Encoder 1.2.0pre3] Surface Submission time: 15.984529
16:39:52.475: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.201627
16:39:52.495: [AMF Encoder 1.2.0pre3] Surface Submission time: 19.802248
16:39:52.498: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.105203
16:39:52.516: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.067180
16:39:52.520: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.249129
16:39:52.539: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.867302
16:39:52.544: [AMF Encoder 1.2.0pre3] Surface Creation time: 4.646854
16:39:52.561: [AMF Encoder 1.2.0pre3] Surface Submission time: 16.759052
16:39:52.564: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.058840
16:39:52.583: [AMF Encoder 1.2.0pre3] Surface Submission time: 19.115331
16:39:52.587: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.358068
16:39:52.604: [AMF Encoder 1.2.0pre3] Surface Submission time: 17.184001
16:39:52.608: [AMF Encoder 1.2.0pre3] Surface Creation time: 3.239742
16:39:52.627: [AMF Encoder 1.2.0pre3] Surface Submission time: 18.975388
16:39:52.630: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.898133
```

It is clearly visible that whatever 16.7.3 did broke Media SDK, as it takes over one frame to submit - something I have no control over.

**Comment by Xaymar** [comment]
After upgrading from a clean slate back to 16.7.3, the submission times are normal again:

```
17:35:27.948: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.637582
17:35:27.950: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.881265
17:35:27.964: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.453836
17:35:27.967: [AMF Encoder 1.2.0pre3] Surface Submission time: 2.071269
17:35:27.981: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.786627
17:35:27.983: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.974845
17:35:27.998: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.373056
17:35:28.000: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.747295
17:35:28.014: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.270090
17:35:28.016: [AMF Encoder 1.2.0pre3] Surface Submission time: 2.200403
17:35:28.031: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.147213
17:35:28.033: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.908855
17:35:28.047: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.381020
17:35:28.050: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.980818
17:35:28.064: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.142662
17:35:28.066: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.656560
17:35:28.081: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.195851
17:35:28.083: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.927912
17:35:28.098: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.336364
17:35:28.100: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.956071
17:35:28.114: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.127871
17:35:28.116: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.881265
17:35:28.131: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.533194
17:35:28.133: [AMF Encoder 1.2.0pre3] Surface Submission time: 2.063588
17:35:28.148: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.434494
17:35:28.150: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.680452
17:35:28.165: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.140671
17:35:28.167: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.967164
17:35:28.181: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.308488
17:35:28.183: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.892358
17:35:28.198: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.162572
17:35:28.200: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.939859
17:35:28.214: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.133560
17:35:28.216: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.859363
17:35:28.231: [AMF Encoder 1.2.0pre3] Surface Creation time: 2.047091
17:35:28.233: [AMF Encoder 1.2.0pre3] Surface Submission time: 1.997599
```

**Comment by Xaymar** [comment]
Updated Issue with solution, has so far worked on all computers I tried it on.

**Comment by Xaymar** [comment]
1.2.0pre3 fixes it partially. Next pre-release/final release will have threaded submission and querying.

**Comment by Xaymar** [comment]
This is theoretically fixed by 1.2.0pre4, if this still occurs for some people make sure that your GPU did not just swap states (reduced or increased clock frequency) while recording.
